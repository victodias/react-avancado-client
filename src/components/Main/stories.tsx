import { Story } from '@storybook/react/types-6-0'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  argTypes: {
    color: { control: 'color' }
  }
  // args: {
  //   title: 'Title default',
  //   description: 'Description default'
  // }
}

export const Basic: Story = (args) => <Main {...args} />
Basic.args = {
  title: 'React AvanÃ§adoo',
  description: 'Typescript, ReactJS, NextJS, e Styled Components',
  color: '#FFF'
}

export const Default: Story = (args) => <Main {...args} />
Default.args = {
  title: 'Default title',
  description: 'Default description',
  color: '#FFF'
}
