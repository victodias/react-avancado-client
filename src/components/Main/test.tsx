import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('Should render the heading', () => {
    const container = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()
    expect(container).toMatchSnapshot('Snapshot of <Main /> with heading')
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
