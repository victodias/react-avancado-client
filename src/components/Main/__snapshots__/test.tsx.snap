// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<Main /> Should render the heading: Snapshot of <Main /> with heading 1`
] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #06092b;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: 'column';
  -ms-flex-direction: 'column';
  flex-direction: 'column';
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
  text-align: center;
  width: 100%;
}

.c1 {
  margin-bottom: 2rem;
  width: 25rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<body>
    <div>
      <main
        class="c0"
      >
        <img
          alt="Imagem de um átomo e React Avançado escrito ao lado."
          class="c1"
          src="/img/logo.svg"
        />
        <h1
          class="c2"
        >
          React Avançado
        </h1>
        <h2
          class="c3"
        >
          Typescript, ReactJS, NextJS, e Styled Components
        </h2>
        <img
          alt="Um desenvolvedor de frente para uma tela com código."
          class="c4"
          src="/img/hero-illustration.svg"
        />
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="styles__Wrapper-sc-1nfdwfb-0 evqmQk"
    >
      <img
        alt="Imagem de um átomo e React Avançado escrito ao lado."
        class="styles__Logo-sc-1nfdwfb-1 hKrChv"
        src="/img/logo.svg"
      />
      <h1
        class="styles__Title-sc-1nfdwfb-2 kxLppZ"
      >
        React Avançado
      </h1>
      <h2
        class="styles__Description-sc-1nfdwfb-3 gbiEmi"
      >
        Typescript, ReactJS, NextJS, e Styled Components
      </h2>
      <img
        alt="Um desenvolvedor de frente para uma tela com código."
        class="styles__Illustration-sc-1nfdwfb-4 kqczPL"
        src="/img/hero-illustration.svg"
      />
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`
